### ISSUE TITLE
[BUG] Pyarmor registration not working due to vEthernet physical address changing

### ISSUE THREAD

**Issue by tagnelli** [dashingsoft/pyarmor#1523]
```
$ pyarmor gen -O dist --assert-call foo.py
INFO     Python 3.11.5
INFO     Pyarmor 8.3.9 (trial), 000000, non-profits
INFO     Platform windows.x86_64
```

Hello, 

I am having an issue when trying to use Pyarmor inside a docker container on Windows.

I am well aware about your documentation. I know you'll probably redirect me to [5.2.1. Check device for group license](https://pyarmor.readthedocs.io/en/latest/licenses.html#check-device-for-group-license) but please, read the following :

We are running Docker on Windows Servercore 2022. The issue we have is that the result of `print(get_hd_info(21))` returns a different value every time despite nothing changing in the machine.

This seems to be the case because by default, docker on Windows create a bridge on the fly that is attached to a virtual ethernet card  that isn't really manageable and **with a different physical address at every boot**.

It is to be noted that a custom **NAT** docker network on Windows [is no longer persisted after reboot.](https://learn.microsoft.com/en-us/virtualization/windowscontainers/container-networking/network-drivers-topologies#nat-network-driver)

So at the moment, it seems we can't use Pyarmor in Docker for Windows with unlimited uses because the host itself will not stay registered after each reboot. 

I believe `get_hd_info()` shouldn't be taking **virtual** network cards into consideration when trying to define hardware ID.

**Comment by jondy** [comment]
Got it. Could you print this NAT interface by example code in this page? 
https://learn.microsoft.com/en-us/windows/win32/api/iptypes/ns-iptypes-ip_adapter_addresses_lh 

Compile the sample code and print all interface information.

**Comment by jondy** [comment]
Please download debug `pytransform3.pyd` to print hardware information

http://pyarmor.dashingsoft.com/downloads/temp/win64/cp3.11/pytransform3.pyd

Then run 
```
python3.11
>>> from pytransform3 import get_hd_info
>>> get_hd_info(21)
```

It should print more information, compare the output after reboot

**Comment by jondy** [comment]
I notice you still use Pyarmor 8.3.9, please upgrade 8.3.10, it fix a group license issue for multiple network interfaces.

Please first try 8.3.10, note that you need generate new device license by 8.3.10, and the machine id should be started with `l` not `k`.

If it still not work, please paste the log output of `pyarmor-auth`, and try the other ways.

**Comment by tagnelli** [comment]
Hi, thanks for coming back to me.
here's the output of the Microsoft script : 
```
Memory allocated for GetAdapterAddresses = 16383 bytes
Calling GetAdaptersAddresses function with family = AF_UNSPEC

        Length of the IP_ADAPTER_ADDRESS struct: 448
        IfIndex (IPv4 interface): 6
        Adapter name: {C293446F-861A-48B2-82F9-1649A95B42AE}
        Number of Unicast Addresses: 1
        No Anycast Addresses
        Number of Multicast Addresses: 3
        Number of DNS Server Addresses: 2
        DNS Suffix:
        Description: Intel(R) PRO/1000 MT Network Connection
        Friendly name: Ethernet
        Physical address: FE-C1-C0-B6-10-50
        Flags: 193
        Mtu: 1500
        IfType: 6
        OperStatus: 1
        Ipv6IfIndex (IPv6 interface): 0
        ZoneIndices (hex): 6 6 6 6 1 1 1 1 1 1 1 1 1 1 0 1
        Number of IP Adapter Prefix entries: 5

        Length of the IP_ADAPTER_ADDRESS struct: 448
        IfIndex (IPv4 interface): 1
        Adapter name: {9C838ED8-6319-11EE-899B-806E6F6E6963}
        Number of Unicast Addresses: 2
        No Anycast Addresses
        No Multicast Addresses
        Number of DNS Server Addresses: 3
        DNS Suffix:
        Description: Software Loopback Interface 1
        Friendly name: Loopback Pseudo-Interface 1
        Flags: 385
        Mtu: 4294967295
        IfType: 24
        OperStatus: 1
        Ipv6IfIndex (IPv6 interface): 1
        ZoneIndices (hex): 1 1 1 1 1 1 1 1 1 1 1 1 1 1 0 1
        Number of IP Adapter Prefix entries: 7

        Length of the IP_ADAPTER_ADDRESS struct: 448
        IfIndex (IPv4 interface): 9
        Adapter name: {A8CF413D-75D6-426F-940D-2ED95A685360}
        Number of Unicast Addresses: 2
        No Anycast Addresses
        Number of Multicast Addresses: 6
        Number of DNS Server Addresses: 3
        DNS Suffix:
        Description: Hyper-V Virtual Ethernet Adapter
        Friendly name: vEthernet (nat)
        Physical address: 00-15-5D-21-66-3F
        Flags: 448
        Mtu: 1500
        IfType: 6
        OperStatus: 1
        Ipv6IfIndex (IPv6 interface): 9
        ZoneIndices (hex): 9 9 9 9 1 1 1 1 1 1 1 1 1 1 0 1
        Number of IP Adapter Prefix entries: 8
```
Here's the result of `get_hd_info(21)` before and after the reboot : 
```
Python 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from pytransform3 import get_hd_info
>>> get_hd_info(21)
01:E               : fec1c0b610P
02:v               : 0015]008ce6
03:disk            : bsjxb5bs7c5yhpvu8i3foi6s8cxwm5anwz
b'l25c9535ebb9a5de8c3758f3d48557a8d'
>>>
```

```
Python 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from pytransform3 import get_hd_info
>>> get_hd_info(21)
01:E               : fec1c0b610P
02:v               : 0015]5c5L
03:disk            : bsjxb5bs7c5yhpvu8i3foi6s8cxwm5anwz
b'l4eb5b6079a359c79b82414fd5424bff5'
```
This was done with pyarmor 8.3.10.

Now that pyarmor is not registered anymore, here's the output of `pyarmor-auth` : 
```
c:\>pyarmor-auth.exe -d .\ProgramData\pyarmor\pyarmor-device-regfile-6083.5.zip
2023-10-12 16:40:27,375: work path: C:\Users\Administrator\.pyarmor\docker
2023-10-12 16:40:27,375: register ".\ProgramData\pyarmor\pyarmor-device-regfile-6083.5.zip"
2023-10-12 16:40:27,390: extracting license.lic
2023-10-12 16:40:27,390: extracting .pyarmor_capsule.zip
2023-10-12 16:40:27,390: group license for machines: ['tokens/l5e241d78c365be6ceeb866236dd9b9c3']
2023-10-12 16:40:27,406: got machine id: l4eb5b6079a359c79b82414fd5424bff5
2023-10-12 16:40:27,406: got machine id: i4eb5b6079a359c79b82414fd5424bff5
2023-10-12 16:40:27,406: got machine id: k9924ffb0c6c96c85f2e62d6c7706bde7
2023-10-12 16:40:27,406: got machine id: g9924ffb0c6c96c85f2e62d6c7706bde7
2023-10-12 16:40:27,406: got machine id: b279db6cc42f3e84babec207237463aba
2023-10-12 16:40:27,422: could not get docker host machine id
Traceback (most recent call last):
  File "c:\Python\Lib\site-packages\pyarmor\cli\register.py", line 198, in _get_docker_hostname
    s.connect((host, port))
socket.gaierror: [Errno 11001] getaddrinfo failed
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "c:\Python\Scripts\pyarmor-auth.exe\__main__.py", line 7, in <module>
  File "c:\Python\Lib\site-packages\pyarmor\cli\docker.py", line 107, in main
    main_entry()
  File "c:\Python\Lib\site-packages\pyarmor\cli\docker.py", line 90, in main_entry
    register_pyarmor(ctx, args.regfile[0])
  File "c:\Python\Lib\site-packages\pyarmor\cli\docker.py", line 62, in register_pyarmor
    reg.register_regfile(regfile)
  File "c:\Python\Lib\site-packages\pyarmor\cli\register.py", line 171, in register_regfile
    raise CliError('could not get docker host machine id')
pyarmor.cli.CliError: could not get docker host machine id
```

To note that pyarmor (gen) and pyarmor-auth work correctly on the machine once registered, until I reboot :smiling_face_with_tear: . 

To note as well that this is running on Windows Server 2022.
Thanks in advance.

**Comment by jondy** [comment]
Got it, thanks for your help.

And you're right, this virtual network is taken into account. It will be fixed in next minor release 8.4.0 because it need upgrade core libraries, this minor release will be released in a couple of weeks to support Python 3.12.

**Comment by jondy** [comment]
Could you download new test extension from
http://pyarmor.dashingsoft.com/downloads/temp/5.4.0/windows.x86_64/libs/cp3.11/pytransform3.pyd

And test machine id changed or not after reboot by this test extension?
```
python3.11
>>> from pytransform3 import get_hd_info
>>> get_hd_info(22)
```

**Comment by jondy** [comment]
Fixed in 8.4.0

Note that it need generate new device file for this device by 8.4.0

And register this new device file also by 8.4.0 to get device group file
